### ISSUE TITLE
removing the foreman_/katello_ prefix from the modules?

### ISSUE THREAD

**Issue by evgeni** [theforeman/foreman-ansible-modules#816]
When used in the collection context, the FQCN is rather long, and often contains "foreman": theforeman.foreman.foreman_host. While the prefix is useful outside of the collection, the duplication inside it is rather annoying.

As we're currently still supporting Ansible 2.7 which has no collection support, we can't really do the change right now. But as soon as 2.10 ships and 2.7 is not supported anymore, I'd like us to rename the modules to drop the prefix.

We could, actually, go and rename them when building the collection already (and keeping the old name as a symlink so that old playbooks keep working).

What do y'all think?

**Comment by evgeni** [comment]
(I've set the milestone to 1.0 as I think the 1.0 release should have stable naming :))

**Comment by ehelms** [comment]
As they were originally how we namespaced the modules, I like dropping the prefix naming on the modules. I assume this would be more consistent with a lot of other collections as well and simplify things.

If it's easy to change it when building the collection: +1

Is it possible to support both naming conventions for some period of time?

**Comment by ehelms** [comment]
Relevant to the conversation is tombstoning that will be in Ansible 2.10:

https://github.com/ansible/ansible/pull/67684

**Comment by evgeni** [comment]
@evgeni will try to whip up a PoC that will have the prefixes stripped but the old names available via deprecation symlinks inside the collection, the repository will remain prefixed for the lifetime of Ansible 2.7

**Comment by ehelms** [comment]
As part of this, or parallel to, we should rename katello_sync to `sync_repository` or something that is less generic than `sync` will be.

**Comment by gundalow** [comment]
ansible-base's routing.yml should point to the new name.

**Comment by evgeni** [comment]
> ansible-base's routing.yml should point to the new name. 

These modules were never in Ansible, so I don't think we need to touch Ansible's routing?

**Comment by gundalow** [comment]
Ah, sorry, ignore my last comment.

**Comment by abenokraitis** [comment]
Hey all, any consensus yet? Maybe a mapping doc (before/after) to help?

**Comment by evgeni** [comment]
@abenokraitis @gundalow may I raise your awareness to https://github.com/ansible/ansible/issues/69902?

TL;DR (also for everyone else): underscore-based deprecation is not supported (and does not work) inside of collections, so while we can (and should) add a `routing` config that will work in Ansible 2.10, there is no way to make it work in 2.8/2.9 today. And even if the routing feature would be backported to older Ansibles, nothing guarantees that users have the latest .z depending on their OS update policies.

Thus, my current proposal would be to just bite the bullet and release a 1.0 that will not redirect the old names on 2.8/2.9.

**Comment by evgeni** [comment]
Ansible 2.10 is [scheduled to be released on 2020-08-13](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_10.rst), which will make 2.7 EOL. Do we want to wait until then, or drop 2.7 support earlier? People could still use a 2.7 compatible branch we can create (similar to what we did with `nailgun`)?

**Comment by ehelms** [comment]
I'm impatient, and don't think there is much overhead in the rename for users. EL7 and EL8 at least all have versions of Ansible that support collections correct? My vote then is for your suggestion of a branch/version that ends 2.7 support and drop it to aim for a 1.0.0 release.

**Comment by abenokraitis** [comment]
you can always add a `requires_ansible: '>=2.9.9,<2.11'` to the `runtime.yml` (was routing.yml) to indicate the downstream collection is only compatible with 2.9.9+ and 2.10.0+

**Comment by evgeni** [comment]
The downstream collection does not have that issue as it has 0 users right now ;)

**Comment by evgeni** [comment]
For Eric's last q: EL7 and EL8 are good, so is Ubuntu 20.04 LTS. Debian 10 has a good version in the backports repository.
